sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -608,7 +608,8 @@ case class FileSourceScanExec(
       fsRelation: HadoopFsRelation): RDD[InternalRow] = {
     val openCostInBytes = fsRelation.sparkSession.sessionState.conf.filesOpenCostInBytes
     val maxSplitBytes =
-      FilePartition.maxSplitBytes(fsRelation.sparkSession, selectedPartitions)
+      FilePartition.maxSplitBytes(fsRelation.sparkSession, selectedPartitions,
+        new SourceOptions(fsRelation.options).parallel)
```

Comment:
I would prefer a hint of parallel rather than in table options, as we prefer using different parallelism when reading data of different sizes.

Reply:
Yes. This is the first step. Will support the hint in following pull request.